### PR TITLE
Rev versions to the next minor+prerelease

### PR DESCRIPTION
**Pull request by RickWinter**
- [ ] The purpose of this PR is explained in this or a referenced issue.
- [ ] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [ ] Tests are included and/or updated for code changes.
- [ ] Updates to module CHANGELOG.md are included.
- [ ] MIT license headers are included in each file.

[Azure/autorest.go]: https://github.com/Azure/autorest.go

### CODE REVIEW

**Review by weshaggard — APPROVED** [review]

**Comment by jhendrixMSFT** [comment]
https://github.com/Azure/azure-sdk-for-go/issues/21799

File: eng/scripts/Update-ModuleVersion.ps1 [hunk 1]
```diff
@@ -40,12 +40,7 @@ if ($incrementVersion) {
     exit 1
   }
 
-  if ($semVer.PrereleaseLabel -ne "zzz") {
-    $semVer.PrereleaseNumber++
-  }
-  else {
-    $semVer.Patch++
-  }
+  $semVer.IncrementAndSetToPrerelease()
```
[the remark this answers is not in the document]
Comment (weshaggard) [reply eng/scripts/Update-ModuleVersion.ps1:43]: See https://github.com/Azure/azure-sdk-for-go/blob/main/eng/common/scripts/SemVer.ps1#L150

**Comment by RickWinter** [comment]
@jhendrixMSFT Any concerns with using the beta tags?

**Comment by jhendrixMSFT** [comment]
It wasn't clear to me how this will work in all cases. i.e. for `v0` we don't want a `-beta.N` suffix.

**Pull request closed without merging**